src/lib/receiver.js

```diff
@@ -186,16 +203,31 @@ function createReceiver (opts) {
     }
 
     // The payment request is extracted from the ilp_header
-    let packet = transfer.data && transfer.data.ilp_header
+    const packet = transfer.data && transfer.data.ilp_header
 
     if (!packet) {
       debug('got notification of transfer with no packet attached')
       return rejectIncomingTransfer(transfer.id, 'no-packet')
     }
 
-    // check if the address contains "~ipr" + our receiver id
-    if (packet.account.indexOf(receiverId) === -1) {
-      debug('got notification of transfer for another receiver: ' + packet.account + ' my id is: ' + receiverId)
+    // check if the address starts with our address
+    if (packet.account.indexOf(getAddress()) !== 0) {
```

Comment:
I'm fine with checking the account. If we make this change though we should remove the option to specify the account in the `createRequest` call (and note that that is technically a breaking change)